packages/builder/src/pages/builder/app/[application]/design/[screenId]/_components/AppPreview.svelte

```diff
@@ -130,6 +131,13 @@
       error = event.error || "An unknown error occurred"
     } else if (type === "select-component" && data.id) {
       componentStore.select(data.id)
+
+      const componentPath = findComponentPath(
+        $selectedScreen?.props,
+        data.id
+      ).map(component => component._id)
+
+      componentTreeNodesStore.expandNodes(componentPath)
```

Comment:
Same as mentioned previously - it would be nice to hide this implementation inside the store and just do `componentTreeNodesStore.revealNode(data.id)`.